src/eth/storage/stratus_storage.rs

```diff
@@ -337,6 +338,31 @@ impl StratusStorage {
         let _span = tracing::info_span!("storage::save_block", block_number = %block.number()).entered();
         tracing::debug!(storage = %label::PERM, block_number = %block.number(), transactions_len = %block.transactions.len(), "saving block");
 
+        let new_block_number = block.number();
+        let last_mined_block_number = self.perm.read_mined_block_number()?;
+
+        // check permanent storage block number
+        if new_block_number != last_mined_block_number + 1 {
+            let gap_size = new_block_number.as_i64() - last_mined_block_number.as_i64();
+            tracing::warn!(
+                ?new_block_number,
+                ?last_mined_block_number,
+                gap_size,
+                "new block number isn't expected based on previous one, some might've been skipped or order is broken",
+            );
+        }
+
+        // check temporary storage block number, if set
+        if let Some(pending_block_number) = self.temp.read_pending_block_number()? {
+            if new_block_number != pending_block_number {
```

Comment:
This check will always be true.
The block being saved is already closed and there is a new pending block being mined.
Probably you want to check that the `new_block_number == (pending_block_number - 1)`.

Example:
Block 1 is closed and being saved. Block 2 is the pending one.

Reply:
oh yeah I did a local test and already fixed it, ty